### ISSUE TITLE
calculation gradients of tf.nn.embedding_lookup

### ISSUE THREAD

**Issue by wweichn** [tensorflow/tensorflow#15084]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**:1.3.0
- **Python version**: 2.7

### Source code / logs
import tensorflow as tf
types_lookup_table = tf.get_variable("types_lookup_table", shape=[234, 10],
                                     initializer=tf.random_normal_initializer(0, 1), dtype=tf.float32,
                                     trainable=True)
embedding_types = tf.nn.embedding_lookup(types_lookup_table,[[2,3,4],[1,2,3]])
opt = tf.train.GradientDescentOptimizer(0.1)
gradients = tf.gradients(embedding_types, xs=types_lookup_table)
train = opt.apply_gradients([(gradients[0], types_lookup_table)])

with tf.Session() as sess:
    tf.global_variables_initializer().run()
    h = sess.run(gradients)
    print(sess.run(train))                                                                              #right
    print(sess.run(opt.apply_gradients([(h[0],types_lookup_table)]))).    # wrong

### Describe the problem
I tried to calculate the gradients of tf.nn.embedding_lookup, but the result shown is an IndexedSliceValue with 3 elements
<img width="1250" alt="2017-12-04 9 23 45" src="https://user-images.githubusercontent.com/10001692/33532464-ec59088e-d8d4-11e7-9c08-d53c6d87abf5.png">
however the corresponding gradient(without sess.run) is an indexSliceValue with 1 elements.I don't know why.
<img width="1226" alt="2017-12-04 9 27 31" src="https://user-images.githubusercontent.com/10001692/33532533-6aa0a5bc-d8d5-11e7-9b0d-69a950cb5fb0.png">

And therefore I can't sess.run(opt.apply_gradients([(h[0],types_lookup_table)]) because the shape of calculation value doesn't match the shape of types_lookup_table, however, when I didn't calculate the intermediate value, and directly 
sess.run(train) (ps:train = opt.apply_gradients([(gradients,types_lookup_table)]))
there is no problem.

But I need to calculate the intermediate value and do an add. I don't know how.
Thanks

**Comment by facaiy** [comment]
Sorry, I cannot figure out what you want.

In your code,
```python
gradients = tf.gradients(embedding_types, xs=types_lookup_table)
h = sess.run(gradients)
print(sess.run(opt.apply_gradients([(h[0],types_lookup_table)]))). # wrong
```
Since `types_lookup_table`'s dimension is 234 * 10, the corresponding `graidents` is a 234 * 10 tensor, as same as `h`. 
So `h[0]` is 1 * 10, while `types_lookup_table` is 234 * 10, it seems wrong to apply `h[0]` to `types_lookup_table` because of the unmatched dimensions.

Please correct me if I'm wrong.

**Comment by wweichn** [comment]
yeah, I though h can be applied to calculate gradients directly.  I mean apply_gradients(h, types_lookup_table). But the size of h is different from gradients in h = sess.run(gradients), however, it seems that h is composed of three elements, and it needs some change to get the final gradients that can be applied to apply_gradients.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?
